### PR TITLE
added chart museum via basic auth

### PR DESCRIPTION
**Pull request by arturrez**
added chart museum via basic http auth as per https://app.zenhub.com/workspaces/cloud-platform-and-infrastructure-5c61e874f2a1351031f66999/issues/upbound/infrastructure-squad/50

### CODE REVIEW

File: makelib/helm.mk [hunk 1]
```diff
@@ -46,6 +46,11 @@ HELM := $(TOOLS_HOST_DIR)/helm-$(HELM_VERSION)
 # remove the leading `v` for helm chart versions
 HELM_CHART_VERSION := $(VERSION:v%=%)
 
+#Chart Museum
```
Comment (jbw976) [inline makelib/helm.mk]: do you want to leave these commented out variables? if so, i'd encourage some more guidance/info about what they are and how they are expected to be used.

note also that this is a public repository, so anything added here should hopefully be generally applicable and relevant.

File: makelib/helm.mk [hunk 2]
```diff
@@ -113,6 +118,16 @@ helm.promote: $(HELM_HOME)
 	@rm -fr $(HELM_TEMP)
 	@$(OK) promoting helm charts
 
+helm.museum: $(HELM_OUTPUT_DIR)/$(1)-$(HELM_CHART_VERSION).tgz
```
Comment (jbw976) [inline makelib/helm.mk]: what exactly is the `$(1)` in this dependency? this one is a bit unintuitive

File: makelib/helm.mk [hunk 3]
```diff
@@ -113,6 +118,16 @@ helm.promote: $(HELM_HOME)
...
+	ifneq ($(MUSEUM_URL),)
+		@$(INFO) pushing helm charts to chart museum
```
Comment (jbw976) [inline makelib/helm.mk]: want to include the museum URL in this INFO statement?

File: makelib/helm.mk (hunk 1 above)
Comment (arturrez) [reply makelib/helm.mk]: added more info on how it can be used

File: makelib/helm.mk (hunk 3 above)
Comment (arturrez) [reply makelib/helm.mk]: done

File: makelib/helm.mk (hunk 2 above)
Comment (arturrez) [reply makelib/helm.mk]: I think I need to redesign the entire thing. it's not that straightforward that I thought

Comment (jbw976) [reply makelib/helm.mk]: can you expand on that thought @arturrez? I'm not sure I have the full context to understand where you think it needs to go.

**Comment by arturrez** [comment]
no I didn't test it even I tested curl command.
looking into it

**Pull request merged**